### DISCUSSION TITLE
cors error while using subdomain

### DISCUSSION THREAD

**Discussion by romanusname** [vercel/next.js#56931] [Help]
### Summary

I created a subdomain. If I'm on http://control.localhost:3000/item/1
Me rewrite http://localhost:3000/control/item/1

**But now every time I call the api I get an error:**
Access to XMLHttpRequest at 'http://localhost:3000/api/v1/items' from origin 'http://control.localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**I've been in pain for a long time now. In my config:**
```
    async headers() {
        return [
            {
                // matching all API routes
                source: "/api/:path*",
                headers: [
                    { key: "Access-Control-Allow-Credentials", value: "true" },
                    { key: "Access-Control-Allow-Origin", value: "*" },
                    {
                        key: "Access-Control-Allow-Methods",
                        value: "GET,OPTIONS,PATCH,DELETE,POST,PUT",
                    },
                    {
                        key: "Access-Control-Allow-Headers",
                        value:
                            "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version",
                    },
                ],
            },
        ];
    },
```

**Please help me find a solution**

### Additional information

```js
nextjs 13 appDir
```

**Comment by iam-joey** [comment]
Did your issue got resolved? can you please tell me how did you rectified it. even i'm also getting the same error
